src/java_bytecode/java_utils.cpp

```diff
@@ -325,3 +326,105 @@ std::string pretty_print_java_type(const std::string &fqn_java_type)
     result = result.substr(java_lang_string.length());
   return result;
 }
+
+/// Finds an inherited component (method or field), taking component visibility
+/// into account.
+/// \param component_class_id: class to start searching from. For example, if
+///   trying to resolve a reference to A.b, component_class_id is "A".
+/// \param component_name: component basename to search for. If searching for
+///   A.b, this is "b".
+/// \param user_class_id: class identifier making reference to the sought
+///   component. The user class is relevant when determining whether package-
+///   scoped components are visible from a particular use site.
+/// \param symbol_table: global symbol table.
+/// \param class_hierarchy: global class hierarchy.
+/// \param include_interfaces: if true, search for the given component in all
+///   ancestors including interfaces, rather than just parents.
+/// \return the concrete component referred to if any is found, or an invalid
+///   resolve_inherited_componentt::inherited_componentt otherwise.
+resolve_inherited_componentt::inherited_componentt get_inherited_component(
```

Comment:
it might make sense to return an optional inherited_componentt here instead of `resolve_inherited_componentt::inherited_componentt();` in the case of access-protected fields.

Reply:
Nah, that type already has a distinguished not-valid value.